### ISSUE TITLE
Date constants are converted to strings by modeler

### ISSUE THREAD

**Issue by pakrym** [Azure/autorest.modelerfour#70]
https://github.com/Azure/autorest.testserver/blob/master/swagger/url.json

```
  - !<!ConstantSchema> &ref_40
    type: constant
    value: !<!ConstantValue> 
      value: '2012-01-01'
      language:
        default:
          name: ''
          description: ''
    valueType: *ref_1
    language: !<!Languages> 
      default:
        name: Constant14
        description: MISSING·SCHEMA-DESCRIPTION-CHOICE
    protocol: !<!Protocols> {}
  - !<!ConstantSchema> &ref_42
    type: constant
    value: !<!ConstantValue> 
      value: '2012-01-01T01:01:01Z'
      language:
        default:
          name: ''
          description: ''
    valueType: *ref_1
    language: !<!Languages> 
      default:
        name: Constant15
        description: MISSING·SCHEMA-DESCRIPTION-CHOICE
    protocol: !<!Protocols> {}
```
```
choiceType: !<!StringSchema> &ref_1
      type: string
      language: !<!Languages> 
        default:
          name: string
          description: simple string
```

**Comment by fearthecowboy** [comment]
I noticed this too. Constants are not checking the extended type.

**Comment by fearthecowboy** [comment]
This is interesting; In v2 we never identified constants early, this was left to late in the generation process. 

In this case, I think I actually have to parse the contents like generated code would have.

**Comment by fearthecowboy** [comment]
Will be addressed by #73

**Comment by fearthecowboy** [comment]
Fixed in #73 --

To use this version:

Ensure that you're on `@autorest/autorest` `3.0.6141` or greater:

`npm install -g @autorest/autorest`
and then do a --reset and get the specific autorest-core package version `3.0.6168`, and the `modelerfour` `4.2.83`

```
autorest-beta --reset 
autorest-beta --version:3.0.6168 --use:@autorest/modelerfour@4.2.83 
```
